src/lib.rs

```diff
@@ -1448,6 +1489,45 @@ impl Build {
                 child.kill().ok();
             }
         }
+
+        #[derive(Default)]
```

Comment:
I know it's not that much of code, but this feels a bit like an implementation detail; I think we should have it in a separate module with an explanation for why we can't just pull an async runtime in and that we have to roll our own async executor.

Reply:
I would definitely prefer we don't pull an async runtime in.

Reply:
Oh yeah, I'm not suggesting that - it's just that at first glance one could ask why we're rolling a custom executor instead of pulling something in. Given how `cc` is at the very bottom of most dependencies graphs, that would be a disaster I guess. =)
All I'm saying is that this decision should be documented to avoid having that discussion in the future.

Reply:
I've added the doc and extract async-executor out as a module as suggested.